build 15.1 testing

rchenatjpl opened the pull request:
<!--
    ************************************** REMINDER **************************************
    PR Titles should be "user-friendly". We use these titles
    to populate our Release Notes. Some examples can be found here:
    https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Issue-Tracking#pull-request-titles
-->

## 🗒️ Summary
Brief summary of changes if not sufficiently described by commit messages.

## ⚙️ Test Data and/or Report
One of the following should be included here:
* Reference to regression test included in code (preferred wherever reasonable)
* Attach test data here + outputs of tests

## ♻️ Related Issues
<!--
    Reference related issues here and use `Fixes` or `Resolves` in order to automatically close the issue upon merge. For more information on autolinking to tickets see https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls.

    * for issues in this repo:
        - fixes #1
        - fixes #2
        - refs #3
    * for issues in other repos: NASA-PDS/my_repo#1, NASA-PDS/her_repo#2
-->

rchenatjpl committed:
build 15.1 testing

rchenatjpl committed:
build 15.1 testing. Hope it overwrites

rchenatjpl committed:
build 15.1 testing. Hope the whole dir goes in

rchenatjpl committed:
build 15.1 testing

rchenatjpl committed:
build 15.1 testing

rchenatjpl committed:
build 15.1 testing

rchenatjpl committed:
build 15.1 testing. File unneeded

rchenatjpl committed:
build 15.1 testing. File unneeded

rchenatjpl committed:
build 15.1 testing

al-niessner reviewed (approved):
I would recommend not using comments in the feature files. The ticket number should be enough to get back to github for the most up to date and complete information. The comments here will simply rot and not be true in a very short period of time.

rchenatjpl committed:
enable content validation in 3.6; comment out failing 3.7 tests

rchenatjpl committed:
Update 1203_12.xml. Change not-matching name

al-niessner commented on src/test/resources/features/3.6.x.feature:
```diff
@@ -18,24 +18,24 @@ Feature: 3.6.x
 | 950 | 2 | "github950" | "--disable-context-mismatch-warnings -R pds4.bundle -t {datasrc}" |  |
 | 919 | 1 | "github919" | "--skip-context-validation -t {datasrc}/uh0003b_draft.xml" |  |
 | 919 | 2 | "github919" | "-t {datasrc}/uh0003b_draft.xml" |  |
-| 915 | 1 | "github915" | "--skip-content-validation --skip-context-validation -R pds4.collection -t {datasrc}/collection.xml" | "summary:totalErrors=1,summary:totalWarnings=1,summary:productValidation:failed=1,summary:messageTypes:error.label.duplicate_identifier=1,summary:messageTypes:warning.integrity.unreferenced_member=1" |
```
Should be fixed with rebase or merge not modified for context here

al-niessner commented on src/test/resources/features/3.6.x.feature:
```diff
@@ -18,24 +18,24 @@ Feature: 3.6.x
 | 905 | 1 | "github905" | "--skip-context-validation -t {datasrc}/dsn_0159-science.2008-02-29.xml {datasrc}/dsn_0159-science.2009-05-18.xml" |  |
 | 905 | 2 | "github905" | "-t {datasrc}/." |  |
 | 902 | 1 | "github902" | "--skip-context-validation -t {datasrc}/s_00168901_thm.xml" |  |
-| 902 | 2 | "github902" | "--skip-content-validation --skip-context-validation -t {datasrc}/s_00168901_thm.xml" |  |
```
Should be fixed with rebase or merge not modified for context here

al-niessner commented on src/test/resources/features/3.6.x.feature:
```diff
@@ -18,24 +18,24 @@ Feature: 3.6.x
 | 861 | 1 | "github861" | "-t {datasrc}/PVO_OMAG_OEFD_ANC_ENG_0001.xml" | "summary:totalWarnings=1,summary:messageTypes:warning.label.context_ref_mismatch=1" |
 | 857 | 1 | "github857" | "-t {datasrc}/highi_183_istria_fam3.xml" | "summary:totalWarnings=2,summary:messageTypes:warning.label.context_ref_mismatch=2" |
 | 849 | 1 | "github849" | "--skip-context-validation -t {datasrc}/collection_uvs_data_raw.xml" | "summary:totalErrors=2,summary:productValidation:failed=1,summary:messageTypes:error.inventory.duplicate_lidvid=2" |
-| 849 | 2 | "github849" | "--skip-context-validation -t {datasrc}/collection.xml" | "summary:totalErrors=1,summary:productValidation:failed=1,summary:messageTypes:error.inventory.duplicate_lidvid=1" |
```
Should be fixed with rebase or merge not modified for context here

al-niessner commented on src/test/resources/features/3.6.x.feature:
```diff
@@ -18,24 +18,24 @@ Feature: 3.6.x
 | 837 | 2 | "github837" | "-t {datasrc}/x.xml" | "summary:totalWarnings=1,summary:messageTypes:warning.label.context_ref_mismatch=1" |
 | 831 | 1 | "github831" | "--skip-context-validation -t {datasrc}/kplo.xml" |  |
 | 824 | 1 | "github824" | "--skip-context-validation -t {datasrc}/1203_12.xml" |  |
-| 824 | 2 | "github824" | "--skip-content-validation --skip-context-validation -t {datasrc}/1203_12.xml" |  |
```
Should be fixed with rebase or merge not modified for context here

al-niessner commented on src/test/resources/features/3.6.x.feature:
```diff
@@ -18,24 +18,24 @@ Feature: 3.6.x
-| 824 | 2 | "github824" | "--skip-content-validation --skip-context-validation -t {datasrc}/1203_12.xml" |  |
+| 824 | 2 | "github824" | "-t {datasrc}/1203_12.xml" |  |
 | 823 | 1 | "github823" | "--skip-context-validation -t {datasrc}/mvn_swi_l2_onboardsvymom_20230827_v02_r01.xml" |  |
-| 822 | 1 | "github822" | "--skip-context-validation -R pds4.bundle -t {datasrc}" | "summary:totalWarnings=1,summary:messageTypes:warning.file.not_referenced_in_label=1" |
```
Should be fixed with rebase or merge not modified for context here

al-niessner commented on src/test/resources/features/3.7.x.feature:
```diff
@@ -7,4 +7,13 @@ Feature: 3.7.x
     Examples: 
       | issueNumber | subtest | datasrc | args | expectation |
 
+#| 992 | 1 | "github992" | "-t {datasrc}/ff_del.xml" | "summary:totalErrors=1,summary:totalWarnings=5,summary:productValidation:failed=1,summary:messageTypes:error.table.field_value_data_type_mismatch=1,summary:messageTypes:warning.table.field_value_format_precision_mismatch=1,summary:messageTypes:warning.table.field_value_format_specifier_mismatch=4,summary:messageTypes:warning.table.field_value ...
```
There is no error error and certainly should not be commented out. Ticket #1179 will either be resolved with an update to the pattern in the PDS4 documentation or determined to be correct as stands. Either way, its original form should remain until #1179 is resolved one way or the other.

jordanpadams replied:
@al-niessner I think this test was copied from the other feature file just to get it started.

al-niessner commented on src/test/resources/features/3.7.x.feature:
```diff
@@ -7,4 +7,13 @@ Feature: 3.7.x
       | issueNumber | subtest | datasrc | args | expectation |
 
+#| 992 | 1 | "github992" | "-t {datasrc}/ff_del.xml" | "summary:totalErrors=1,summary:totalWarnings=5,summary:productValidation:failed=1,summary:messageTypes:error.table.field_value_data_type_mismatch=1,summary:messageTypes:warning.table.field_value_format_precision_mismatch=1,summary:messageTypes:warning.table.field_value_format_specifier_mismatch=4,summary:messageTypes:warning.table.field_value ...
+#| 1028 | 1 | "github1028" | "-t {datasrc}" |  |
+#| 1066 | 1 | "github1066" | "-t {datasrc}" |  |
```
If you added these (I see test content for this ticket) and you changed your mind, then delete all of it. If you modified it and changed your mind, then it should be reverted to its original form or pick up fixes from a merge or rebase.

al-niessner reviewed (commented):
Adding tests is awesome and thank you so much. If this ticket is to be merged back into main, then some of these changes are spurious and should be undone. Turning off tests is never the right answer because it makes validate vulnerable to bug regression - the whole point of testing.

rchenatjpl committed:
Update 3778ml1037770010808163i01_dxxx.xml

Changed the name associated with
  urn:nasa:pds:context:instrument_host:spacecraft.msl
from
  MARS SCIENCE LABORATORY
to
  The Mars Science Laboratory Curiosity Rover
